core/src/bitcoin/BitcoinLikeExtendedPublicKey.hpp

```diff
@@ -31,28 +31,30 @@
 #ifndef LEDGER_CORE_BITCOINLIKEEXTENDEDPUBLICKEY_HPP
 #define LEDGER_CORE_BITCOINLIKEEXTENDEDPUBLICKEY_HPP
 
-#include "../api/BitcoinLikeExtendedPublicKey.hpp"
-#include "../crypto/DeterministicPublicKey.hpp"
-#include "../api/BitcoinLikeNetworkParameters.hpp"
 #include <memory>
-#include "../utils/Option.hpp"
-#include "../utils/DerivationPath.hpp"
+#include <common/AbstractExtendedPublicKey.h>
+#include <api/BitcoinLikeExtendedPublicKey.hpp>
+#include <crypto/DeterministicPublicKey.hpp>
+#include <api/BitcoinLikeNetworkParameters.hpp>
+#include <utils/Option.hpp>
+#include <utils/DerivationPath.hpp>
 #include <api/Currency.hpp>
 
 namespace ledger {
     namespace core {
-        class BitcoinLikeExtendedPublicKey : public api::BitcoinLikeExtendedPublicKey {
+        using BitcoinExtendedPublicKey = AbstractExtendedPublicKey<api::BitcoinLikeNetworkParameters>;
+        class BitcoinLikeExtendedPublicKey : public BitcoinExtendedPublicKey, public api::BitcoinLikeExtendedPublicKey {
         public:
             BitcoinLikeExtendedPublicKey(const api::Currency& params,
                                          const DeterministicPublicKey& key, const DerivationPath& path = DerivationPath("m/"));
-            virtual std::shared_ptr<api::BitcoinLikeAddress> derive(const std::string &path) override;
-            virtual std::shared_ptr<BitcoinLikeExtendedPublicKey> derive(const DerivationPath& path);
+            std::shared_ptr<api::BitcoinLikeAddress> derive(const std::string &path) override;
+            std::shared_ptr<BitcoinLikeExtendedPublicKey> derive(const DerivationPath& path);
```

Comment:
Hmm I don't see a reason why do that, we don't have any implicit conversion on `DerivationPath` since its string based ctor is explicit ... unless you have another argument that I missed ? 🤔